### DISCUSSION TITLE
What is the list of languages that will definitely have long-term support in the Meson build system?

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#12476] [Q&A] [answered]
Is Meson build planning on keeping support for Cython? I'm looking into adding new templates for the Meson build generator. In addition to this question, what is the list of languages that will definitely have long-term support in the Meson build system?

**Comment by eli-schwartz** [comment, marked as answer]
In general, dropping a language would require first deprecating it and then waiting until a new major release (e.g. waiting until meson 2.0). We'd also have to have a good reason why we no longer consider it tenable to support that language and want to drop it.

Cython is very important to us, since we have major stakeholders using it that are maintaining an affiliated @mesonbuild repository (meson-python). Even if we weren't interested in actively maintaining cython support, they would be more than willing to personally keep it working.

In short, Cython is here to stay.

**Reply by eli-schwartz** [reply]
If there was a language I'd like to get rid of, it would be vala :rofl: because vala is either dead or on life support depending on who you ask, and they transpile to C code which modern compilers don't like or reject altogether.

But there are no actual plans to drop even that.

**Reply by dreamer-coding** [reply]
I never really understood the purpose of Vala

**Reply by nda-cunh** [reply]
all vala projects use meson, and vala is an excellent language...
Vala is very interesting for Linux and gtk applications

**Reply by Kangie** [reply]
> all vala projects use meson, and vala is an excellent language...

Except the part where they can't transpile to C code that modern compilers will accept.
